src/utility.ts

```diff
@@ -215,6 +215,18 @@ export function getLauncherScriptPath() {
     return path.join(ext.extensionPath, "scripts", "launcher.bat");
 }
 
+export function isGitBash(isIntegratedTerminal: boolean): boolean {
+    const terminal: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("terminal");

```

Comment:
try vscode's API for current shell?
`const currentWindowsShellPath: string = vscode.env.shell;`

Reply:
yes, we can use this API to get current shell for integrated terminal.